python/metatensor-torch/setup.py

```diff
@@ -228,6 +228,38 @@ def create_version_number(version):
 
 
 if __name__ == "__main__":
+    if sys.platform == "win32":
+        # On Windows, starting with PyTorch 2.3, the file shm.dll in torch has a
+        # dependency on mkl DLLs. When building the code using pip build isolation, pip
+        # installs the mkl package in a place where the os is not trying to load
+        #
+        # This is a very similar fix to https://github.com/pytorch/pytorch/pull/126095,
+        # except only applying when importing torch from a build-isolation virtual
+        # environment created by pip (`python -m build` does not seems to suffer from
+        # this).
```

Comment:
We'll need to keep this hack around until https://github.com/pytorch/pytorch/pull/131340 is merged, release, and available on all torch releases we support.